### PR TITLE
[Enhancement] Support PredicateTree for parquet page index

### PR DESCRIPTION
**Pull request by Smith-Cruise**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.4
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0

### CODE REVIEW

File: be/src/storage/rowset/segment_iterator.cpp [hunk 1]
```diff
@@ -45,6 +45,7 @@
 #include "storage/index/vector/vector_index_reader_factory.h"
 #include "storage/index/vector/vector_search_option.h"
 #include "storage/lake/update_manager.h"
+#include "storage/olap_runtime_range_pruner.h"
```
Comment (Smith-Cruise) [inline be/src/storage/rowset/segment_iterator.cpp:48]: fix compile problem in clang-18

File: be/src/formats/parquet/scalar_column_reader.cpp [hunk 2]
```diff
@@ -28,16 +29,24 @@ StatusOr<bool> FixedValueColumnReader::row_group_zone_map_filter(const std::vect
...
+    ZoneMapDetail zone_map{_fixed_value, _fixed_value, _fixed_value.is_null()};
 
-    return is_satisfy(zone_map);
+    if (ZoneMapEvaluatorUtils::is_satisfy(predicates, zone_map, pred_relation)) {
+        // no filter happened
+        return false;
+    } else {
+        // filter happened, entire row group can be filtered
+        return true;
+    }
```
Comment (zombee0) [inline be/src/formats/parquet/scalar_column_reader.cpp:49]: return !ZoneMapEvaluatorUtils::is_satisfy(predicates, zone_map, pred_relation);

**Review by zombee0 — COMMENTED** [review]
could you check if all the data has been filtered, will the predicate tree return immediately?

File: be/src/formats/parquet/scalar_column_reader.cpp (hunk 2 above)
Comment (Smith-Cruise) [reply be/src/formats/parquet/scalar_column_reader.cpp]: done

**Comment by Smith-Cruise** [comment]
> could you check if all the data has been filtered, will the predicate tree return immediately?

Of course it can

**Review by zombee0 — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Pull request merged**